R/get_vimc_climate.R

```diff
@@ -0,0 +1,150 @@
+#' Get climate data Parquets prepared for VIMC
+#'
+#' @param country A string for the country name, or for the country ISO 2- or 3-
+#' character code. E.g. one of "Canada", "CA", or "CAN". Input is checked
+#' against names and codes in \pkg{countrycode}.
+#' @param date_range A two element vector of either strings or `<Date>`s, giving
+#' the interval for which to get data. If a character vector, must be in a
+#' format that can be converted to a `<Date>` by [lubridate::as_date()].
+#' @param data_location A location relative to the current working directory
+#' to search for the data. Data for `country` are expected to live in
+#' `data_location/XYZ` as Parquet files, where `XYZ` is the ISO 3-character code
+#' for `country`. Defaults to the working directory.
+#' @param data_source A string giving the climate data to search for. Names must
+#' match exactly.
+#' @param admin_level A numeric giving the GADM admin level for which to get
+#' data. Only one unit may be passed at a time.
+#'
+#' @return A `<data.frame>` of climate data.
+#'
+#' @export
+get_vimc_climate <- function(
```

Comment:
I have a real dislike of function names starting with `get_` - it seems both too much of an implementation detail (like pointless getters and setters in OOP) and too little - is this loading from a local file or is it fetching from a remote resource. It would be nice to sketch out somewhere the possible total package API in a ticket so that we get a nice set of functions that hint at their functionality

Reply:
It's a load function for now, but the idea is to have a single function that can access different types of locations, both local dirs and `packit`. Is there something specific to rename this to?